apps/nerves_hub_web_core/lib/nerves_hub_web_core/devices/ca_certificate.ex

```diff
@@ -22,8 +22,32 @@ defmodule NervesHubWebCore.Devices.CACertificate do
     :last_used
   ]
 
+  defmodule JITP do
+    use Ecto.Schema
+    import Ecto.Changeset
+    alias NervesHubWebCore.Products.Product
+
+    schema "jitp" do
+      # JITP enabled CA Certs must be linked to a product
+      # so devices can be created
+      belongs_to(:product, Product)
+
+      # Will be coppied to the device when it's created
+      field(:tags, NervesHubWebCore.Types.Tag)
+      field(:description, :string)
+      timestamps()
+    end
+
+    def changeset(jitp, params) do
+      jitp
+      |> cast(params, [:tags, :description])
+      |> validate_required([])
+    end
+  end
+
   schema "ca_certificates" do
     belongs_to(:org, Org, where: [deleted_at: nil])
+    belongs_to(:jitp, JITP)
```

Comment:
```suggestion
    has_one(:jitp, JITP)
```

Should this be a `has_one`? Otherwise, but `ca_certificate` and `jitp` belong to each other with seems circular?

Reply:
jitp doesn't `belong_to` jitp here. `has_one` is the opposite of `belongs_to` where `has_one` doesn't actually use any database records if i remember correctly. the `ca_certificates` table has a column called `jitp_id` but the `jitp` table does **not** have a `ca_certificate_id`. I'm pretty sure this is "normal" for this kind of thing, but i've also been known to get it backwards.